faros_event.sh

```diff
@@ -402,432 +345,280 @@ function resolveDefaults() {
+            {
+                "sha": $commit_sha,
+                "repository": $commit_repo,
+                "organization": $commit_org,
+                "source": $commit_source
+            }' <<< $request_body
+        )
+    fi
 }
 
-function addCIObjectsToEvent() {
-    makeArtifact
-    makeArtifactCommitAssociation
-    request_body=$(jq ".entries += [
-        $cicd_Artifact,
-        $cicd_ArtifactCommitAssociation
-    ]" <<< $request_body)
-}
+function addArtifactToData() {
+    if ! [ -z "$artifact_source" ] &&
+       ! [ -z "$artifact_org" ] &&
+       ! [ -z "$artifact_repo" ] &&
+       ! [ -z "$artifact_id" ]; then
+        request_body=$(jq \
+            --arg artifact_source "$artifact_source" \
+            --arg artifact_org "$artifact_org" \
+            --arg artifact_repo "$artifact_repo" \
+            --arg artifact_id "$artifact_id" \
+            '.data.artifact +=
+            {
+                "source": $artifact_source,
+                "organization": $artifact_org,
+                "repository": $artifact_repo,
+                "id": $artifact_id
 
-function addCICDObjectsToEvent() {
-    makeOrganization
-    makePipeline
-    request_body=$(jq ".entries += [
-        $cicd_Organization,
-        $cicd_Pipeline
-    ]" <<< $request_body)
+            }' <<< $request_body
+        )
+    fi
 }
 
-function addCICDRepositoryToEvent() {
-    makeArtifactRepository
-    request_body=$(jq ".entries += [
-        $cicd_Repository
-    ]" <<< $request_body)
+function addRunToData() {
+    if ! [ -z "$run_source" ] &&
+       ! [ -z "$run_org" ] &&
+       ! [ -z "$run_pipeline" ] &&
+       ! [ -z "$run_id" ]; then
+        request_body=$(jq \
+            --arg run_source "$run_source" \
+            --arg run_org "$run_org" \
+            --arg run_pipeline "$run_pipeline" \
+            --arg run_id "$run_id" \
+            '.data.run +=
+            {
+                "source": $run_source,
+                "organization": $run_org,
+                "pipeline": $run_pipeline,
+                "id": $run_id
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_status" ]; then
+        request_body=$(jq \
+            --arg run_status "$run_status" \
+            '.data.run +=
+            {
+                "status": $run_status
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_status_details" ]; then
+        request_body=$(jq \
+            --arg run_status_details "$run_status_details" \
+            '.data.run +=
+            {
+                "statusDetails": $run_status_details,
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_start_time" ]; then
+        request_body=$(jq \
+            --arg run_start_time "$run_start_time" \
+            '.data.run +=
+            {
+                "startTime": $run_start_time|tonumber
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$run_end_time" ]; then
```

Comment:
for all the time values - are these ISO8601 string **or** millis since epoch numerical value? or we support both?

Reply:
Right now the CLI only supports millis because of jq converting the string that is passed to a number... It throws a parsing error if its not a valid number.

Reply:
Got it. Just trying to understand, why were you adding iso time support in the event API?

Reply:
I just added Millisecond time support to the Event API so that the CLI could switch over with minimal changes necessary for people to upgrade. As to why we support ISO time in the API, we wanted that to be the new standard that we ask people to provide for events (I think we are also hoping that at some point the graphQL api also converts time into ISO format too, so it's human readable iirc).

Reply:
I got confused! Makes sense.

We now could offer support for ISO8601 timestamps in Events CLI.

Reply:
(no need to do it in this PR)

Reply:
Yeah, good call. I do want to support ISO8601 timestamps in the CLI. Let me see what I can do.